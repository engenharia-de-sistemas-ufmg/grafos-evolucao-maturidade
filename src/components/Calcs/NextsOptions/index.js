import { connect } from 'react-redux';
import NextsOptions from './NextsOptions';
import { hideNexts } from '../../../actions/nexts';

const mapStateToProps = (state) => ({
  nexts: state.nexts,
});

const mapDispatchToProps = (dispatch) => ({
  hideNexts: () => { dispatch(hideNexts()); },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(NextsOptions);
