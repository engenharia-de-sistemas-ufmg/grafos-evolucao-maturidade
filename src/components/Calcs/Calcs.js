import React from 'react';
import { nextsType } from '../../utils/types';
import NextsOptions from './NextsOptions/index';

const Calcs = ({
  nexts,
}) => {
  if (nexts.visible) {
    return (<NextsOptions />);
  }
  return null;
};

Calcs.propTypes = {
  nexts: nextsType.isRequired,
};

export default Calcs;
