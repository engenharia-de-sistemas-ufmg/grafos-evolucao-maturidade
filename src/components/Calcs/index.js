import { connect } from 'react-redux';
import Calcs from './Calcs';

const mapStateToProps = (state) => ({
  nexts: state.nexts,
});

export default connect(
  mapStateToProps,
  null,
)(Calcs);
