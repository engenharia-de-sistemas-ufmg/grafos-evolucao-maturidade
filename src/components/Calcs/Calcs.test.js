import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Calcs from './Calcs';
import basicGraph from '../../examples/basicGraph';

describe('Calcs', () => {
  Enzyme.configure({ adapter: new Adapter() });

  test('should match snapshot', () => {
    const nexts = {
      calculated: false,
      calculating: false,
      options: [],
    };
    const component = shallow(
      <Calcs
        lang="pt"
        graph={basicGraph}
        showNexts={() => { }}
        nexts={nexts}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
