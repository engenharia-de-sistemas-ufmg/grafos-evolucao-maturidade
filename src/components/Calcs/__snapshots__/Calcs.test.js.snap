// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calcs should match snapshot 1`] = `""`;
